[ci] Add ghstack /land bot!!!

poteto opened the pull request:
Adds a new `/land` command that can be written as a comment on a pull request. The command must be the very first line of the comment, like so:

```
/land

<additional context etc>
```

The workflow will first check if the commenter is a collaborator or member, and additionally also check if the commenter is a maintainer via the MAINTAINERS file.

The workflow will then attempt to validate the pull request, checking that CI has completed successfully and that it has received at least one approval before landing. The land is performed via `ghstack land`, which does mean that the PR itself isn't merged directly via github but it is pushed to main by a synthetic user (@facebook-github-bot for now). This means PRs landed with `/land` will have an additional co-author @facebook-github-bot, but the original committer will not be lost.

poteto commented:
/land

rickhanlonii commented on .github/workflows/shared_ghstack_land.yml:
```diff
@@ -0,0 +1,121 @@
+      actor: ${{ github.event.comment.user.login }}
+
+  ghstack_land:
+    if: ${{ needs.check_maintainer.outputs.is_core_team == 'true' && github.event.issue.pull_request && contains(github.event.comment.body, '/land') }}
```
maybe startsWith or only? lf something includes a URL with /land in it that seems bad

poteto replied:
good call, fixed

poteto commented on .github/workflows/shared_ghstack_land.yml:
```diff
@@ -0,0 +1,121 @@
+            const repo = context.repo.repo;
+            const comment_id = "${{ github.event.comment.id }}"
+
+            await github.rest.reactions.createForCommitComment({
```
nope, only a small handful of endpoints are now GraphQL only. Most of them are still REST based.

rickhanlonii commented on .github/workflows/shared_ghstack_land.yml:
```diff
@@ -0,0 +1,121 @@
+      - name: Land It!
+        run: |
+          git config --global user.email "facebook-github-bot@users.noreply.github.com"
+          git config --global user.name "Facebook Community Bot"
```
Does this break the commit verification? 

Ideally this would be set this to the user that does the /land command, I believe that's what a merge would do. But that would break the verification too.

poteto committed:
[ci] Add ghstack /land bot

Adds a new `/land` command that can be written as a comment on a pull request. The command must be the very first line of the comment, like so:

```
/land

<additional context etc>
```

The workflow will first check if the commenter is a collaborator or member, and additionally also check if the commenter is a maintainer via the MAINTAINERS file.

The workflow will then attempt to validate the pull request, checking that CI has completed successfully and that it has received at least one approval before landing. The land is performed via `ghstack land`, which does mean that the PR itself isn't merged directly via github but it is pushed to main by a synthetic user (@facebook-github-bot for now). This means PRs landed with `/land` will have an additional co-author @facebook-github-bot, but the original committer will not be lost.
